File: Julia/sort/selection_sort.jl
# Julia program to implement Selection sort

function selection_sort(arr, n)
    for i in 1:n
        # Let the first element in this iteration be the minimum value
        min = i
        # Find the least element in the concerned part of the array
        for j in (i + 1):n
            if arr[min] > arr[j]
                min = j
            end
        end
        # Swap the least element with the beginning of the concerned part of the array
        arr[i], arr[min] = arr[min], arr[i]
    end
end



print("How many numbers do you want to sort? ")
n = readline()
n = parse(Int, n)
if (n <= 0)
    println("No numbers to sort!!!")
    exit()
end
arr = Int[]
print("Enter the numbers: ")
arr = [parse(Int, num) for num in split(readline())]
selection_sort(arr, n)
print("The numbers in sorted order is ")
print(arr)



"""
Time Complexity - O(n^2), where 'n' is the size of the array
Space Complexity - O(n)

SAMPLE INPUT AND OUTPUT

SAMPLE I
How many numbers do you want to sort? 5
Enter the numbers: 4 2 5 1 3
The numbers in sorted order is [1, 2, 3, 4, 5]

"""

File: Python/sort/Selection_Sort.py
#Python program to implement Selection Sort

def selectionSort(array, n):
    i = 0
    while i < n:
        #Initilizing mini as i
        mini = i
        j = i + 1
        while j < n:
            #If element at j is smaller than mini then change mini to j
            if array[j] < array[mini]:
                mini = j
            j += 1
        #swapping elements at mini and i    
        (array[i], array[mini]) = (array[mini], array[i])
        i += 1          


data = []
n = int(input("Enter the size of array :"))
 
print('Enter the elements of array:');
for i in range(n):
    data.append(int(input()))
    
# Sending element to get sorted
selectionSort(data, n)
 
# Priting Elements after Getting Sorted
for d in data:
    print(d, end = " ")

"""
Time Complexity - O(n^2), where 'n' is the size of the array
Space Complexity - O(1)

SAMPLE INPUT AND OUTPUT

SAMPLE I

INPUT
Enter the size of array: 5
Enter the elements of array: 
5
4 
3 
2 
1

OUTPUT
1 2 3 4 5

"""